lib/view_component/base.rb

```diff
@@ -159,8 +159,12 @@ def compiled?
         @compiled && ActionView::Base.cache_template_loading
       end
 
+      def inlined?
+        instance_methods(false).grep(/^call/).present? && templates.empty?
+      end
+
       def compile!
-        compile(raise_template_errors: true)
+        compile(raise_template_errors: true) unless inlined?
```

Comment:
Would it be possible to move `unless inlined?` to after `return if compiled?` below? That way we won't run this check at runtime if we've compiled the component.

Reply:
Let me know if that works.